src/gossip.rs

```diff
@@ -133,28 +133,17 @@ impl GossipSource {
 }
 
 pub(crate) struct RuntimeSpawner {
-	runtime: Arc<RwLock<Option<Arc<tokio::runtime::Runtime>>>>,
-	logger: Arc<Logger>,
+	runtime: Arc<Runtime>,
 }
 
 impl RuntimeSpawner {
-	pub(crate) fn new(
-		runtime: Arc<RwLock<Option<Arc<tokio::runtime::Runtime>>>>, logger: Arc<Logger>,
-	) -> Self {
-		Self { runtime, logger }
+	pub(crate) fn new(runtime: Arc<Runtime>) -> Self {
+		Self { runtime }
 	}
 }
 
 impl FutureSpawner for RuntimeSpawner {
```

Comment:
Can this be implemented directly onto the new `Runtime`?

Reply:
Is there really no way to do this? Implement an interface on a local type and then pass it to the other crate? Or would it require a different type on the LDK side for FutureSpawner?

Perhaps longer term moving `Runtime` to rust-lightning could be a direction too?